# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1522750541 -7200
#      Tue Apr 03 12:15:41 2018 +0200
# Branch stable
# Node ID dd480b75e6e8df8f5fcc6ff2d5350a205a21df5a
# Parent  078b795c52191c929a960da45ed0ecaccf2ee898
octave.css: Provide fallback monospace fonts (bug #53006).

* octave.css: add 'Courier New', Cousine, and Courier fallback fonts in
case 'Fantasque Sans Mono' is not installed and the doc is being rendered offline.

* doc/interpreter/module.mk: Add a dependency on HTMLDIR_CSS to
OCTAVE_QTHELP_FILES so they are rebuilt when styles change.

diff --git a/doc/interpreter/module.mk b/doc/interpreter/module.mk
--- a/doc/interpreter/module.mk
+++ b/doc/interpreter/module.mk
@@ -199,17 +199,17 @@ HTMLDIR_CSS = $(OCTAVE_HTML_DIR)/octave.
 
 if AMCOND_BUILD_QT_DOCS
 
 OCTAVE_QTHELP_FILES = %reldir%/octave_interpreter.qhc %reldir%/octave_interpreter.qch
 
 octdoc_DATA += \
   $(OCTAVE_QTHELP_FILES)
 
-$(OCTAVE_QTHELP_FILES): $(OCTAVE_HTML_STAMP) %reldir%/mk-qthelp.pl
+$(OCTAVE_QTHELP_FILES): $(OCTAVE_HTML_STAMP) $(HTMLDIR_CSS) %reldir%/mk-qthelp.pl
 	$(AM_V_GEN)rm -f $(OCTAVE_QTHELP_FILES) && \
 	$(PERL) $(srcdir)/%reldir%/mk-qthelp.pl octave.html %reldir%/octave_interpreter && \
 	$(QCOLLECTIONGENERATOR) $(QCOLLECTIONGENERATORFLAGS) %reldir%/octave_interpreter.qhcp -o %reldir%/octave_interpreter.qhc >/dev/null && \
 	rm -f %reldir%/octave_interpreter.qhcp %reldir%/octave_interpreter.qhp
 
 ## The Qt help collection generator command produces two output files
 ## with the same base name: the compressed help (qch) file and the help
 ## collection (qhc) file.  Declare the qhc file to depend on the
diff --git a/doc/interpreter/octave.css b/doc/interpreter/octave.css
--- a/doc/interpreter/octave.css
+++ b/doc/interpreter/octave.css
@@ -57,56 +57,56 @@ along with Octave; see the file COPYING.
        url('https://fontlibrary.org/assets/fonts/roboto-condensed/71405335c70332d94afd24ae4f06c9b2/b0b1845ecce8ab6a503971e808a8409c/RobotoCondensedRegular.ttf') format('truetype');
   font-weight: normal;
   font-style: normal;
 }
 
 /* Import web font Fantasque Sans Mono, Font Library Version 2016-04-05 */
 @font-face
 {
-  font-family: 'Fantasque Sans Mono';
+  font-family: 'Fantasque Sans Mono', 'Courier New', Cousine, Courier, monospace;
   src: local('Fantasque Sans Mono Bold'),
        local('FantasqueSansMono-Bold'),
        url('https://fontlibrary.org/assets/fonts/fantasque-sans-mono/b0cbb25e73a9f8354e96d89524f613e7/a46033d3a07d9385620dc83b7655203f/FantasqueSansMonoBold.ttf') format('truetype');
   font-weight: bold;
   font-style: normal;
 }
 
 @font-face
 {
-  font-family: 'Fantasque Sans Mono';
+  font-family: 'Fantasque Sans Mono', 'Courier New', Cousine, Courier,  'monospace';
   src: local('Fantasque Sans Mono Bold Italic'),
        local('FantasqueSansMono-BoldItalic'),
        url('https://fontlibrary.org/assets/fonts/fantasque-sans-mono/b0cbb25e73a9f8354e96d89524f613e7/b0683ef4c834908aa372ec78dea42349/FantasqueSansMonoBoldItalic.ttf') format('truetype');
   font-weight: bold;
   font-style: italic;
 }
 
 @font-face
 {
-  font-family: 'Fantasque Sans Mono';
+  font-family: 'Fantasque Sans Mono', 'Courier New', Cousine, Courier, monospace;
   src: local('Fantasque Sans Mono Italic'),
        local('FantasqueSansMono-Italic'),
        url('https://fontlibrary.org/assets/fonts/fantasque-sans-mono/b0cbb25e73a9f8354e96d89524f613e7/b78d0a48e4443f797e8d45dcf97594f7/FantasqueSansMonoItalic.ttf') format('truetype');
   font-weight: normal;
   font-style: italic;
 }
 
 @font-face
 {
-  font-family: 'Fantasque Sans Mono';
+  font-family: 'Fantasque Sans Mono', 'Courier New', Cousine, Courier, monospace;
   src: local('Fantasque Sans Mono Regular'),
        local('FantasqueSansMono-Regular'),
        url('https://fontlibrary.org/assets/fonts/fantasque-sans-mono/b0cbb25e73a9f8354e96d89524f613e7/f3fdc4f0f26e4431f54e8b552d55480c/FantasqueSansMonoRegular.ttf') format('truetype');
   font-weight: normal;
   font-style: normal;
 }
 
 /* Use custom fonts */
-code, kbd, samp, tt, pre { font-family: 'Fantasque Sans Mono', monospace; }
+code, kbd, samp, tt, pre { font-family: 'Fantasque Sans Mono', 'Courier New', Cousine, Courier, monospace; }
 
 code, kbd, samp, tt {
   font-style: italic;
   padding: 0 0.1ex; /* slightly increase margin to surrounding text */ }
 
 body, .sansserif { font-family: 'Roboto Condensed', sans-serif; }
 
 h1, h2, h3, h4, h5, .menu-comment, .roman, .menu-preformatted {
